test/Atmos/EDMF/helper_funcs/diagnose_environment.jl

```diff
@@ -0,0 +1,72 @@
+#### Diagnose environment variables
+
+"""
+    environment_vars(state::Vars, aux::Vars, N_up::Int)
+
+A NamedTuple of environment variables
+"""
+function environment_vars(state::Vars, aux::Vars, N_up::Int)
+    return (
+        a = environment_area(state, aux, N_up),
+        w = environment_w(state, aux, N_up),
+    )
+end
+
+"""
+    environment_area(
+        state::Vars,
+        aux::Vars,
+        N_up::Int,
+    )
+
+Returns the environmental area fraction, given:
+ - `state`, state variables
+ - `aux`, auxiliary variables
+ - `N_up`, number of updrafts
+"""
+function environment_area(state::Vars, aux::Vars, N_up::Int)
+    up = state.turbconv.updraft
+    return 1 - sum(vuntuple(i -> up[i].ρa, N_up)) / state.ρ
```

Comment:
I don't think so. The area fraction limit should be enforced in the updraft stage. If we diagnose env<0, something is wrong in the prognostic model.

Reply:
True. We should still find a good place to check for this, however. Probably in `subdomain_thermo_states.jl`, since this is one of the earliest places where `a_en` is computed. The other reason that `subdomain_thermo_states.jl` is a good place is that we also need to check the bounds on `q_tot_en`. I'll add some error handling there.

Reply:
For a single updraft the EDMF equations do not know that updraft must be smaller or equal to 1 , 
and at the same way for multiple updrafts the equations do not know that sum{I}{updraft area} should be smaller or equal to 1. Both should be enforced. until we have a better way to do it from the sources (entr-detr) it should be done directly on the area fractions. 
I agree that the enforcement should be on the updrafts rather than the environment, but in the case of mutiny updrafts that is a bit harder and hard to know which updraft to clip

Reply:
I added some error handling. Checking `a_min <= a_en <= a_max` seems to fail, so I left this out for now.